plugin/storage/es/spanstore/reader.go

```diff
@@ -91,15 +93,19 @@ type SpanReader struct {
 	serviceOperationStorage *ServiceOperationStorage
 }
 
-// NewSpanReader returns a new SpanReader.
-func NewSpanReader(client es.Client, logger *zap.Logger, maxSpanAge time.Duration) *SpanReader {
+// NewSpanReader returns a new SpanReader with a metrics.
+func NewSpanReader(client es.Client, logger *zap.Logger, maxSpanAge time.Duration, metricsFactory metrics.Factory) spanstore.Reader {
+	return storageMetrics.NewReadMetricsDecorator(newSpanReader(client, logger, maxSpanAge), metricsFactory)
+}
+
+func newSpanReader(client es.Client, logger *zap.Logger, maxSpanAge time.Duration) *SpanReader {
 	ctx := context.Background()
 	return &SpanReader{
 		ctx:                     ctx,
 		client:                  client,
 		logger:                  logger,
 		maxSpanAge:              maxSpanAge,
-		serviceOperationStorage: NewServiceOperationStorage(ctx, client, logger, 0),
+		serviceOperationStorage: NewServiceOperationStorage(ctx, client, metrics.NullFactory, logger, 0), // the decorator takes care of metrics
```

Comment:
the ServiceOperationStorage is common to both reader and writer, but only the writer needs the metrics factory, because the reader uses a decorator instead for metrics